File: Bubble Sort/BubbleSort.java

public class BubbleSort {

  public static void bubbleSort(int[] array) {
    int arrLength = array.length;

    for (int i = 0; i < arrLength - 1; i++) {
      for (int j = 0; j < arrLength - i - 1; j++) {
        if (array[j] > array[j + 1]) {
          int temp = array[j];
          array[j] = array[j + 1];
          array[j + 1] = temp;
        }
      }
    }
  }
    

  public static void main(String[] args) {
    
    int[] array = {
      5, 1, 11, 0, 2, 15, 4, 3, 9, 
    };
    
    bubbleSort(array);
    
    for (int num : array) {
            System.out.print(num + " ");
        }
  }

}

File: Bubble Sort/BubbleSort.py
def bubble_sort(arr):
    arr_length = len(arr)

    for i in range(arr_length - 1):
        for j in range(arr_length - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

if __name__ == "__main__":
    array = [5, 1, 11, 0, 2, 15, 4, 3, 9]
    bubble_sort(array)

    print("Sorted Array:", array)
